week3/homework/src/index.js

```diff
@@ -13,38 +13,63 @@ const {
   markAsNotDone
 } = require('./actions');
 
-// app.post('/todos', (req, res) {
-//   createTodo();
-// })
 app.post('/todos', (req, res) => {
-  createTodo(req, res);
+  createTodo(req, res)
+    .then(todos => res.status(200).json(todos))
+    .catch(error => res.status(500).send({ message: error.message }));
 });
 
 app.get('/todos', (req, res) => {
   readTodos(req, res)
     .then(todos => res.status(200).json(todos))
-    .catch(error => res.status(500).send("can't read file!", error));
+    .catch(error =>
+      res.status(500).send({ message: "can't read file!", error })
+    );
 });
 
 app.put('/todos/:id', (req, res) => {
-  updateTodo();
+  updateTodo(req, res)
+    .then(todos => res.status(200).json(todos))
+    .catch(error => res.status(500).send({ message: error.message }));
 });
 
 app.delete('/todos/:id', (req, res) => {
-  deleteTodo();
+  deleteTodo(req, res)
+    .then(todo => res.status(200).json(todo))
```

Comment:
Even though there is more code now, I think it is a huge improvement with good error handling and nice error messages. Good job 👍